### PR TITLE
opencl: extended OpenCL support for basecurve module (fusion)

### PR DESCRIPTION
**Pull request by upegelow**
This brings quite a performance improvement even on low-end GPUs.

@hanatos: please review

### CODE REVIEW

File: src/iop/basecurve.c [hunk 1]
```diff
@@ -284,8 +299,389 @@ void init_presets(dt_iop_module_so_t *self)
...
+    dt_opencl_set_kernel_arg(devid, gd->kernel_basecurve_finalize, 3, sizeof(int), (void *)&width);
+    dt_opencl_set_kernel_arg(devid, gd->kernel_basecurve_finalize, 4, sizeof(int), (void *)&height);
+    err = dt_opencl_enqueue_kernel_2d(devid, gd->kernel_basecurve_finalize, sizes);
+    if(err != CL_SUCCESS) goto error;
+  } while(0);
+
+
+  for(int k = 0; k < num_levels_max; k++)
+  {
+    dt_opencl_release_mem_object(dev_col[k]);
```
Comment (LebedevRI) [inline src/iop/basecurve.c]: Hm, so what happens if it is `NULL`?

**Comment by upegelow** [comment]
As discussed: in real life nothing.

**Comment by LebedevRI** [comment]
So `clReleaseMemObject()` is okay with `NULL`, and checking for non-`NULL` before calling `dt_opencl_release_mem_object()` is completely unneeded?

**Comment by upegelow** [comment]
As discussed: the specs don't explicitely tell what happens for NULL. 
But all real life implementations seem to accept it.

I will take care of that in due time.

**Comment by LebedevRI** [comment]
Ok, just checking because i see no `if(ptr == null) return;` in `dt_opencl_release_mem_object()` yet, and no `NULL` checks here.

**Comment by LebedevRI** [comment]
I'm guessing 5d96d761ef283fc8ef28e6c6905047f2973e637e caused merge conflicts here.

**Pull request merged**